### ISSUE TITLE
Enable unitTests run on OSX (Travis) and Windows (AppVeyor)

### ISSUE THREAD

**Issue by piponazo** [Exiv2/exiv2#191]
While working on #189 to reenable the OSX jobs on travis-ci I had to disable the unit tests after detecting some linking issues. They are also disabled on AppVeyor because of similar reasons.

We should enable the unit tests on those platforms as soon as possible.

**Comment by clanmills** [comment]
Are you sure, Luis?  It's linking and running the test suite on both my machines.

There's something weird going on with exiv2-test  I'm aware that it sometimes fails.  I started to investigate  exiv2-test on Monday, then it seemed to be OK again.

**Comment by piponazo** [comment]
Yes, you can see here the travis output:

https://travis-ci.org/piponazo/exiv2/jobs/315786486

There errors were:

```
[ 56%] Linking CXX executable ../bin/unit_tests

Undefined symbols for architecture x86_64:

  "testing::internal::EqFailure(char const*, char const*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool)", referenced from:

      testing::AssertionResult testing::internal::CmpHelperEQFailure<int, int>(char const*, char const*, int const&, int const&) in test_types.cpp.o

      testing::AssertionResult testing::internal::CmpHelperEQFailure<int, long>(char const*, char const*, int const&, long const&) in test_types.cpp.o

      testing::AssertionResult testing::internal::CmpHelperEQFailure<unsigned int, unsigned int>(char const*, char const*, unsigned int const&, unsigned int const&) in test_tiffheader.cpp.o

      testing::AssertionResult testing::internal::CmpHelperEQFailure<int, unsigned short>(char const*, char const*, int const&, unsigned short const&) in test_tiffheader.cpp.o

      testing::AssertionResult testing::internal::CmpHelperEQFailure<Exiv2::ByteOrder, Exiv2::ByteOrder>(char const*, char const*, Exiv2::ByteOrder const&, Exiv2::ByteOrder const&) in test_tiffheader.cpp.o

      testing::AssertionResult testing::internal::CmpHelperEQFailure<int, int>(char const*, char const*, int const&, int const&) in test_futils.cpp.o

ld: symbol(s) not found for architecture x86_64

clang: error: linker command failed with exit code 1 (use -v to see invocation
```

I am sure that it is something quick to fix, but it is very slow to try things on travis-ci and the OSX images ;(

**Comment by clanmills** [comment]
Ah.  The linking issue is in the unit test code.  I have to get "up to speed" with what's going on with all the wonderful stuff you and @D4N have contributed.

My heart always sinks when I see that message **_ld: symbol(s) not found for architecture x86_64_**  It appears to say "somewhere in the swamp you are linking a 32 bit library".  MacOSX hasn't supported 32 bit code since 2012. 

I think the message is only saying "Entry point not found".  I find the command _nm -g --demangle foo.dylib_ useful for inspecting the library.  I don't see the entry points:
```
601 rmills@rmillsmbp:~/gnu/github/exiv2/exiv2/build $ nm -g --demangle lib/libexiv2.dylib  | grep EqFailure
602 rmills@rmillsmbp:~/gnu/github/exiv2/exiv2/build $ nm -g --demangle lib/libexiv2.dylib  | grep Factory
00000000000b5140 T Exiv2::ImageFactory::open(std::__1::auto_ptr<Exiv2::BasicIo>)
...
603 rmills@rmillsmbp:~/gnu/github/exiv2/exiv2/build $ 
```
The obvious fix for the sluggish travis/MacOSX performance is to get Jenkins running again.  I was about 1 hour away from fixing that in September and ran out of time (had to run to Heathrow to fly to Boston).  I'm "more or less" locked out of GitHub at the moment (by a combination of my lack of knowledge of Git and the need to have PRs approved and merged by another team member).

Can you change a setting to enable me to approve my own changes on GitHub.com?

**Comment by piponazo** [comment]
I am not sure what you meant by your last question, can you give me more details about this in slack ? :)

**Comment by D4N** [comment]
The error on OSX looks like it is caused by gtest not being properly compiled. I took a look at the travis output and the build for gtest throws the following warnings:
```
clang: warning: libstdc++ is deprecated; move to libc++ [-Wdeprecated]

ld: warning: directory not found for option '-L/Users/travis/conanData/gtest/1.8.0/lasote/stable/build/8a078c48948419ab3d0002124caf659302a11603/_build/googletest-release-1.8.0/googlemock/gtest/src'
```
So my guess is, that the conanfile is the problem here as it links against the wrong C++ standard library (and then symbols are missing). That might be also the problem on Windows.

**Comment by piponazo** [comment]
Yes, I thought also about that and I wanted to check with `libc++`. In the case of windows is a bit different. I will try to use in AppVeyor VS2017 instead of VS2015, so that I have exactly the same configuration in AppVeyor than in my local machine.

**Comment by piponazo** [comment]
Resolved in #195
